gax-java/gax/src/main/java/com/google/api/gax/retrying/ExponentialRetryAlgorithm.java

```diff
@@ -232,6 +229,15 @@ public boolean shouldRetry(TimedAttemptSettings nextAttemptSettings) {
     return true;
   }
 
+  // For non-LRO RPCs, do not attempt to retry if the totalTime spend is over
+  // the totalTimeout (timeout is in the past) or at the totalTimeout (timeout
+  // will occur immediately). For any other value, the deadlineScheduler will
+  // terminate in the future (even if the timeout is small).
+  @InternalApi
+  protected boolean shouldRPCTerminate(long timeLeftMs) {
```

Comment:
Yeah, this is here due the the very *subtle* difference between unary/streaming vs LRO retry logic. Polling from LRO's should retry even if timeLeftMs == 0 (as it would be the final polling attempt), but unary/streaming callables should not (and it would signify an rpc timeout attempt of 0)